sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -110,38 +110,44 @@ case class CreateViewCommand(
 
   private def isTemporary = viewType == LocalTempView || viewType == GlobalTempView
 
-  // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS' to be consistent with 'CREATE TEMPORARY TABLE'
-  if (allowExisting && isTemporary) {
-    throw new AnalysisException(
-      "It is not allowed to define a TEMPORARY view with IF NOT EXISTS.")
-  }
+  if(isTemporary) verifyTempView()
+
+  private def verifyTempView(): Unit = {
+    // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS'
+    // to be consistent with 'CREATE TEMPORARY TABLE'
+    if (allowExisting) {
+      throw new AnalysisException(
+        "It is not allowed to define a TEMPORARY view with IF NOT EXISTS.")
+    }
 
-  // Temporary view names should NOT contain database prefix like "database.table"
-  if (isTemporary && name.database.isDefined) {
-    val database = name.database.get
-    throw new AnalysisException(
-      s"It is not allowed to add database prefix `$database` for the TEMPORARY view name.")
+    // Temporary view names should NOT contain database prefix like "database.table"
+    if (name.database.isDefined) {
```

Comment:
nit:
```scala
    name.database.foreach { database =>
      throw new AnalysisException(
        s"It is not allowed to add database prefix `$database` for the TEMPORARY view name.")
    }
```

Reply:
`name.database` returns String.

Reply:
It's not, it's an `Option[String]`, I imagine, in which case this is indeed a little more idiomatic.

Reply:
My bad. `name` is an `Option[String]`.

Reply:
Now, I see how `foreach` is working here.
Thanks @MaxGekk